src/Platforms/CPU/CMakeLists.txt

```diff
@@ -2,14 +2,17 @@
 #// This file is distributed under the University of Illinois/NCSA Open Source License.
 #// See LICENSE file in top directory for details.
 #//
-#// Copyright (c) 2020 QMCPACK developers.
+#// Copyright (c) 2023 QMCPACK developers.
 #//
 #// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
 #//
 #// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
 #//////////////////////////////////////////////////////////////////////////////////////
 
-add_library(platform_cpu_runtime INTERFACE)
+# To enable a custom qmcplusplus::isnan, remove fast-math. Only the file scope CMAKE_CXX_FLAGS gets affected.
+string(REPLACE " -ffast-math" "" CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
```

Comment:
Do all compilers use a flag with the same name to perform the optimization that affects isnan?

Reply:
All the compilers that have `-ffast-math` behave the same. I explicitly tested nvhpc which doesn't support `-ffast-math` and passes the added tests.

Reply:
In any case this PR still expands the range of cases for which we can catch a NaN. Hopefully the code path won't get much use...